### PR TITLE
Editor Changes

### PR DESCRIPTION
**Pull request by DarrachBarneveld**
# ✨ Codu Pull Request 💻

Codu Logo

👉 _**Please remove the below and replace with your own values, leaving the headers where they are.**_ 👈

## Pull Request details:
Added target blank and code highlight styling changes. Having issues rendering styled code to preview post. Install lowlight/highlight packages

## Any Breaking changes:
None

Assosiation with issue #498 and #497

### CODE REVIEW

File: components/editor/editor/CodeBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        {extension.options.lowlight
+          .listLanguages()
+          .map((lang: string, index: number) => (
+            <option key={index} value={lang}>
+              {lang}
+            </option>
+          ))}
+      </select>
+
+      <pre className="bg-black border ">
```
Comment (xiaoniuniu89) [inline components/editor/editor/CodeBlock.tsx]: @NiallJoeMaher are there styles for code blocks at the moment? may be easier to add here.

File: components/editor/editor/CodeBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { NodeViewContent, NodeViewWrapper } from "@tiptap/react";
+import { ChangeEvent, FunctionComponent } from "react";
+
+// Change this for code styling
+import "highlight.js/styles/monokai-sublime.css";
```
Comment (xiaoniuniu89) [inline components/editor/editor/CodeBlock.tsx]: changes look good, check with @NiallJoeMaher re styling of the code blocks. Good job.

File: components/editor/editor/CodeBlock.tsx (hunk 1 above)
Comment (xiaoniuniu89) [reply components/editor/editor/CodeBlock.tsx]: Happy to deal with the smaller style issues at the end. Its not super important for alpha page

File: components/editor/editor/CodeBlock.tsx (hunk 2 above)
Comment (DarrachBarneveld) [reply components/editor/editor/CodeBlock.tsx]: I had quite a lot of difficulty getting the syntax highlighter to work in preview/render mode. Will take another look now

Comment (xiaoniuniu89) [reply components/editor/editor/CodeBlock.tsx]: https://tiptap.dev/guide/node-views#render-html

you can extend the configuration for the node in tiptap extensions - . eg Codeblock.configure{renderHTML({ HTMLAttributes }) {
  return ['my-custom-node', mergeAttributes(HTMLAttributes)]
},}

Its not essential now, we can fix in another pr

Comment (DarrachBarneveld) [reply components/editor/editor/CodeBlock.tsx]: I will try this, I went another way thats probably not ideal

Comment (DarrachBarneveld) [reply components/editor/editor/CodeBlock.tsx]: The problem is its parses as a different node. The StarterKit.Codeblock one, not the customised one, 🤔

Comment (NiallJoeMaher) [reply components/editor/editor/CodeBlock.tsx]: Is there any way to extend our current code blocks or is it just too different?

File: components/editor/editor/extensions/index.tsx [hunk 3]
```diff
@@ -156,3 +185,8 @@ export const TiptapExtensions = [
     allowFullscreen: true,
   }),
 ];
+
+function highlightCode(code, language) {
+  const result = lowlight.highlight(language, code);
+  return `<pre><code class="hljs ${language}">${result}</code></pre>`;
```
Comment (xiaoniuniu89) [inline components/editor/editor/extensions/index.tsx:191]: Do we have to use this hljs class, can we copy our styles from our current code block style? 

I am happy to fix styles at a later date in any case.

@NiallJoeMaher

File: components/editor/editor/CodeBlock.tsx (hunk 2 above)
Comment (xiaoniuniu89) [reply components/editor/editor/CodeBlock.tsx]: you should be able to configure all of that. But If what you have works, then we can add it. I don't mind to fix styles up at the end.

File: components/editor/editor/CodeBlock.tsx (hunk 1 above)
Comment (NiallJoeMaher) [reply components/editor/editor/CodeBlock.tsx]: The styles we currently have are great; I wonder if we can mimic or what people think 🤔

File: components/editor/editor/CodeBlock.tsx (hunk 2 above)
Comment (xiaoniuniu89) [reply components/editor/editor/CodeBlock.tsx]: technically everything can be target in global.css with .ProseMirror code, .ProseMirror p etc to make the styles sync - we can figure it out later if you wish, one source of truth should be feasable in the editors case

Comment (DarrachBarneveld) [reply components/editor/editor/CodeBlock.tsx]: I have two solutions. Maybe I should PR the alternate also, each have pros and cons I think?

File: components/editor/editor/CodeBlock.tsx (hunk 1 above)
Comment (xiaoniuniu89) [reply components/editor/editor/CodeBlock.tsx]: We should be able to - We can add this and figure it out - if it doesnt work we can revert the changes and save syntax highlighting for v2 of the editor

File: components/editor/editor/CodeBlock.tsx (hunk 2 above)
Comment (NiallJoeMaher) [reply components/editor/editor/CodeBlock.tsx]: It might be worth chatting about it in the Discord too just so we can discuss too!

Comment (DarrachBarneveld) [reply components/editor/editor/CodeBlock.tsx]: Ok, will log on this evening 👍

**Comment by NiallJoeMaher** [comment]
Let me know if you are going to fix the conflicts and we can add it in case we need backward compatibility.

**Comment by NiallJoeMaher** [comment]
If you can would you fix the conflicts and I'll get this merged! I also need to send you the Codú Hacktoberfest badge 🦡

**Comment by DarrachBarneveld** [comment]
OMG! A collectible badge, my weakness. Will fix conflicts now 😃

**Review by NiallJoeMaher — APPROVED** [review]
LGTM! 🌮

**Pull request merged**